tensorflow-ops/src/TensorFlow/Logging.hs

```diff
@@ -0,0 +1,158 @@
+-- Copyright 2016 TensorFlow authors.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- | TensorBoard Summary generation. Provides type safe wrappers around raw
+-- string emitting CoreOps.
+--
+-- Example use:
+--
+-- > -- Call summary functions while constructing the graph.
+-- > createModel = do
+-- >   loss <- -- ...
+-- >   TF.scalarSummary loss
+-- >
+-- > -- Write summaries to an EventWriter.
+-- > train = TF.withEventWriter "/path/to/logs" $ \eventWriter -> do
+-- >     summaryTensor <- TF.build TF.allSummaries
+-- >     forM_ [1..] $ \step -> do
+-- >         if (step % 100 == 0)
+-- >             then do
+-- >                 ((), summaryBytes) <- TF.run (trainStep, summaryTensor)
+-- >                 let summary = decodeMessageOrDie (TF.unScalar summaryBytes)
+-- >                 TF.logSummary eventWriter step summary
+-- >             else TF.run_ trainStep
```

Comment:
The interface feels a bit awkward. I wonder if we could add `SummaryTensor` as a separate type instead of an alias of `Tensor Value ByteString` and then handle the decoding automatically.